src/sshfs_mount/sshfs_mount.cpp

```diff
@@ -26,37 +26,93 @@
+const std::string snap_path_key{"SNAP="};
+
+auto run_cmd(mp::SSHSession& session, std::string&& cmd)
 {
     auto ssh_process = session.exec(cmd);
     if (ssh_process.exit_code() != 0)
-        error_handler(ssh_process);
+        throw std::runtime_error(ssh_process.read_std_error());
+
     return ssh_process.read_std_output();
 }
 
-auto run_cmd(mp::SSHSession& session, std::string&& cmd)
+auto match_line(std::istringstream& output, const std::string& matcher)
 {
-    auto error_handler = [](mp::SSHProcess& proc) { throw std::runtime_error(proc.read_std_error()); };
-    return run_cmd(session, std::forward<std::string>(cmd), error_handler);
+    output.seekg(0, output.beg);
+
+    std::string line;
+    while (std::getline(output, line, '\n'))
+    {
+        if (line.find(matcher) != std::string::npos)
+        {
+            return line;
+        }
+    }
+
+    return std::string{};
 }
 
-void check_sshfs_exists(mp::SSHSession& session)
+auto get_sshfs_exec_and_options(mp::SSHSession& session)
 {
-    auto error_handler = [](mp::SSHProcess& proc) {
-        mpl::log(mpl::Level::warning, category,
-                 fmt::format("Unable to determine if 'sshfs' is installed: {}", proc.read_std_error()));
-        throw mp::SSHFSMissingError();
-    };
-
-    run_cmd(session, "which sshfs", error_handler);
+    std::string sshfs_exec;
+
+    try
+    {
+        // Prefer to use Snap package version first
+        std::istringstream sshfs_env{run_cmd(session, "sudo multipass-sshfs.env")};
+
+        auto ld_library_path = match_line(sshfs_env, ld_library_path_key);
+        auto snap_path = match_line(sshfs_env, snap_path_key);
+        snap_path = snap_path.substr(snap_path_key.length());
+
+        sshfs_exec = fmt::format("{} {}/bin/sshfs", ld_library_path, snap_path);
+    }
+    catch (const std::exception& e)
+    {
+        mpl::log(mpl::Level::debug, category, fmt::format("'sshfs' snap package is not installed: {}", e.what()));
+
+        // Fallback to looking for distro version if snap is not found
+        try
+        {
+            sshfs_exec = run_cmd(session, "sudo which sshfs");
+        }
+        catch (const std::exception& e)
+        {
+            mpl::log(mpl::Level::warning, category,
+                     fmt::format("Unable to determine if 'sshfs' is installed: {}", e.what()));
+            throw mp::SSHFSMissingError();
+        }
+    }
+
+    sshfs_exec = mp::utils::trim_end(sshfs_exec);
+
+    std::string fuse_version;
+    std::istringstream version_info{run_cmd(session, fmt::format("sudo {} -V", sshfs_exec))};
+
+    auto fuse_version_line = match_line(version_info, fuse_version_string);
+    if (!fuse_version_line.empty())
+    {
+        fuse_version = fuse_version_line.substr(fuse_version_string.length());
+    }
+
+    sshfs_exec += " -o slave -o transform_symlinks -o allow_other";
+
+    // The option was removed in libfuse 3.0
+    if (fuse_version < "3")
```

Comment:
Should we `atoi` this?

Reply:
Well, `atoi` would probably work in this case, but I do see a need that we may need to break down the version string into tokens and then compare each x, y, and z parts of the version.

For example, a new feature may be added in the minor version of `libfuse` or even `sshfs`.  So I think I should change this into a util to figure out the version we are interested in.

Reply:
We already have `semver`?

Reply:
Yep, switched to using `semver`.